### PR TITLE
Adding error message for InputParameter groups

### LINKED ISSUE #6475: InputParameters does not check if the parameter to be added in a group is valid
### Description of the enhancement or error report
As title. I noticed this because I removed one parameter but forgot removing it from the `addParamNamesToGroup` list.

### Identified impact
(i.e. Internal object changes, limited interface changes, public API change, or a list of specific applications impacted)

This issue is minor because it just results into a null operation. It just makes the code looks bad.

### PR DESCRIPTION
**Pull request by permcody**
closes #6475

### CODE REVIEW

**Comment by permcody** [comment]
@YaqiWang - This was your ticket and sure enough you have a few of these invalid parameters hanging out in yak!

**Comment by YaqiWang** [comment]
Oh, good to catch them ;-) Thanks!

**Comment by YaqiWang** [comment]
Ahha, I see moose also contains some invalid params ;-)

**Comment by YaqiWang** [comment]
I checked out your branch and now rebuilding to see if I have fixed all. @permcody Will let you know once it is fixed.

**Comment by permcody** [comment]
Moose had several. Let me know and thanks

**Comment by YaqiWang** [comment]
@permcody yak should have been fixed. I invalidated the test. Btw bugs ON yak should be more proper hehe.

File: framework/src/utils/InputParameters.C [hunk 1]
```diff
@@ -524,9 +524,17 @@ InputParameters::addParamNamesToGroup(const std::string & space_delim_names, con
...
+  // to get a raw list of parameter names to compare against.
+  std::set<std::string> param_names;
+  for (InputParameters::iterator it = begin(); it != end(); ++it)
+    param_names.insert(it->first);
 
+  for (std::vector<std::string>::const_iterator it = elements.begin(); it != elements.end(); ++it)
+    if (param_names.find(*it) != param_names.end())
+      _group[*it] = group_name;
+    else
+      mooseError("Unable to find a parameter with name: " << *it << " when adding to group " << group_name << '.');
```
Comment (jwpeterson) [inline framework/src/utils/InputParameters.C]: Nice accidental insertion fix!

**Pull request merged**